pkg/sink/reconciler/streamsink/streamsink.go

```diff
@@ -95,7 +95,8 @@ func (r *Reconciler) updateTLSSecret(ctx context.Context, secret *corev1.Secret)
 	tlsSecret, err := GetTLSSecret(secret.Data)
 	if err != nil {
 		logging.FromContext(ctx).Errorw("Error reading TLS configuration", zap.Error(err))
+	} else {
```

Comment:
nit: to make the code a bit cleaner here, instead of using an `else`, could you `return` in the `if err != nil` block?